src/app/main.rs

```diff
@@ -499,9 +500,27 @@ where
                 .layer(metrics::layer::<_, classify::Response>(retry_http_metrics))
                 .layer(insert::target::layer());
 
-            let balancer_stack = svc::builder()
+            let balancer = svc::builder()
                 .layer(balance::layer(EWMA_DEFAULT_RTT, EWMA_DECAY))
-                .layer(resolve::layer(Resolve::new(resolver)))
+                .layer(resolve::layer(Resolve::new(resolver)));
+
+            let fallback = svc::builder()
+                .layer(
+                    router::layer(|req: &http::Request<_>| {
+                        let ep = outbound::Endpoint::from_orig_dst(req);
+                        debug!("outbound ep={:?}", ep);
+                        ep
+                    })
+                    .configured(router::Config::new(
```

Comment:
This is necessary as the router layer cannot make a `Service<Config>` here --- we need the primary and fallback layers to take the same target types.

I've considered refactoring router construction so that all router layers to be constructed with a config rather than binding to it after the stack is built, but that would change a lot of unrelated code & ought to be a separate branch. This works for now.

Reply:
Can we just move all of this into the `router::layer` constructor? Is there a good reason to have two versions of the router layer, really?  Perhaps that should be its own setup change (to change router configuration to use a Target type of `()` and to require the layer have the required configuration).

Reply:
@olix0r I made the change you suggested in #253, and rebased this branch to be on top of that branch. Thanks!